src/GraphQL.MicrosoftDI/ConnectionResolverBuilder.cs

```diff
@@ -0,0 +1,405 @@
+        /// </summary>
+        public ConnectionResolverBuilder<TSourceType, TReturnType, T1, T2> WithServices<T1, T2>()
+            => new ConnectionResolverBuilder<TSourceType, TReturnType, T1, T2>(_builder, _scoped);
+
+        /// <inheritdoc cref="WithServices{T1, T2}"/>
+        public ConnectionResolverBuilder<TSourceType, TReturnType, T1, T2, T3> WithServices<T1, T2, T3>()
+            => new ConnectionResolverBuilder<TSourceType, TReturnType, T1, T2, T3>(_builder, _scoped);
+
+        /// <inheritdoc cref="WithServices{T1, T2}"/>
+        public ConnectionResolverBuilder<TSourceType, TReturnType, T1, T2, T3, T4> WithServices<T1, T2, T3, T4>()
+            => new ConnectionResolverBuilder<TSourceType, TReturnType, T1, T2, T3, T4>(_builder, _scoped);
+
+        /// <inheritdoc cref="WithServices{T1, T2}"/>
+        public ConnectionResolverBuilder<TSourceType, TReturnType, T1, T2, T3, T4, T5> WithServices<T1, T2, T3, T4, T5>()
+            => new ConnectionResolverBuilder<TSourceType, TReturnType, T1, T2, T3, T4, T5>(_builder, _scoped);
+
+        /// <summary>
+        /// Specifies that the resolver should run within its own dependency injection scope.
+        /// </summary>
+        public ConnectionResolverBuilder<TSourceType, TReturnType> WithScope()
+        {
+            _scoped = true;
+            return this;
+        }
+
+        /// <summary>
+        /// Specifies the return data type of the connection resolver. Typically this would be <see cref="Connection{TNode}"/>
+        /// or <see cref="Connection{TNode, TEdge}"/>, but it can be any type that returns the proper fields for the
+        /// connection graph type.
+        /// <br/><br/>
+        /// For instance, to specify a return type for a Widget you might write:
+        /// <br/><br/>
+        /// <code>
+        /// .Returns&lt;Connection&lt;Widget&gt;&gt;()
+        /// </code>
+        /// </summary>
+        public ConnectionResolverBuilder<TSourceType, TNewReturnType> Returns<TNewReturnType>()
+            => new ConnectionResolverBuilder<TSourceType, TNewReturnType>(_builder, _scoped);
+
+        /// <summary>
+        /// Specifies the delegate to execute when the field is being resolved.
+        /// </summary>
+        public void Resolve(Func<IResolveConnectionContext<TSourceType>, TReturnType> resolver)
+        {
+            if (_scoped)
+                _builder.ResolveScoped(resolver);
+            else
+                _builder.Resolve(context => resolver(context));
+        }
+
+        /// <inheritdoc cref="Resolve(Func{IResolveConnectionContext{TSourceType}, TReturnType})"/>
+        public void ResolveAsync(Func<IResolveConnectionContext<TSourceType>, Task<TReturnType>> resolver)
+        {
+            if (_scoped)
+                _builder.ResolveScopedAsync(resolver);
+            else
+                _builder.ResolveAsync(async context => await resolver(context));
+        }
```

Comment:
```suggestion
        => _scoped ? _builder.ResolveScopedAsync(resolver) : _builder.ResolveAsync(resolver);
```

Reply:
The same suggestions for other builders in this file.

Reply:
![image](https://user-images.githubusercontent.com/6377684/112333817-4c1c4600-8c91-11eb-9fc3-e182267a7216.png)

Reply:
Also cannot reduce `.ResolveAsync(async context => await resolver(context))` to `.ResolveAsync(resolver)` because the `Resolve` method expects a `Task<object>`, not a `Task<T>`.

Reply:
With #2458 this is now able to be completed and has been done.